### PR TITLE
Vector DB secure connections

### PR DESCRIPTION
**Pull request by gaya3-zipstack**
## What

Making sure that all vector DB connections are secure using SSL. 
Enable SSL for Postgres connections as an explicit option

## Why

Postgres vector db connections need to be explicitly enabled for SSL/TLS communication

## How

By exposing an option in UI to enable/disable SSL. This will help user to connect to locally hosted Postgres withut SSL and remote connections using SSL.

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions / Env Variables

- NA

## Notes on Testing

Tested all vector DB connection to ensure we have SSL enabled. Only for Postgres, it had to be added explicitly.

## Screenshots

UI
image

Postgres Remote (Always TLS connection)
Without SSL
With TLS filter, no packets are captured. Errors out
image

Without TLS filter, packets are captured
image

With SSL
With TLS capture filter, packets are captured
image

Postgres Local 

Without SSL
No capture filter. Packets are captured
image

With TLS capture filter, no packets are captured
image

Local with SSL
With TLS filter, no packer captured.
image

No capture filter. Packets captured
image

Pinecone - Serverless
image

Pinecone - Pod
image

Weaviate
image

Quadrant
image

Milvus
image

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: src/unstract/sdk/adapters/vectordb/postgres/src/static/json_schema.json [hunk 1]
```diff
@@ -41,6 +41,11 @@
       "title": "Password",
       "format": "password",
       "default": ""
+    },
+    "enable_ssl": {
+      "type": "boolean",
+      "title": "Enable SSL",
+      "default": true
```
Comment (muhammad-ali-e) [inline src/unstract/sdk/adapters/vectordb/postgres/src/static/json_schema.json:48]: please put descreption also

**Review by muhammad-ali-e — APPROVED** [review]
LGTM, left a comment

**Review by kirtimanmishrazipstack — APPROVED** [review]
LGTM

File: src/unstract/sdk/adapters/vectordb/postgres/src/static/json_schema.json (hunk 1 above)
Comment (gaya3-zipstack) [reply src/unstract/sdk/adapters/vectordb/postgres/src/static/json_schema.json:49]: Thanks @muhammad-ali-e. Addressed.

**Review by muhammad-ali-e — APPROVED** [review]
LGTM

**Pull request merged**